crates/graphql/src/schema/objects/wallet.rs

```diff
@@ -41,17 +41,51 @@ impl Wallet {
     }
 
     pub async fn profile(&self, ctx: &AppContext) -> FieldResult<Option<TwitterProfile>> {
-        let twitter_handle = self.twitter_handle.clone();
-
-        if twitter_handle.is_none() {
-            return Ok(None);
-        }
-
-        let twitter_handle = twitter_handle.unwrap();
+        let twitter_handle = match self.twitter_handle {
+            Some(ref t) => t.clone(),
+            None => return Ok(None),
+        };
 
         ctx.twitter_profile_loader
             .load(twitter_handle)
             .await
             .map_err(Into::into)
     }
+
+    pub fn connection_counts(&self) -> FieldResult<ConnectionCounts> {
+        Ok(ConnectionCounts {
+            address: self.address.clone(),
+        })
+    }
+}
+
+pub struct ConnectionCounts {
+    pub address: PublicKey<Wallet>,
+}
+
+#[graphql_object(Context = AppContext)]
+impl ConnectionCounts {
+    pub fn from_count(&self, ctx: &AppContext) -> FieldResult<i32> {
+        let db_conn = ctx.db_pool.get()?;
+
+        let count: i64 = graph_connections::table
+            .filter(graph_connections::from_account.eq(&self.address))
+            .count()
+            .get_result(&db_conn)
+            .context("Failed to count from_connections")?;
+
+        Ok(count.try_into()?)
+    }
+
+    pub fn to_count(&self, ctx: &AppContext) -> FieldResult<i32> {
+        let db_conn = ctx.db_pool.get()?;
+
+        let count: i64 = graph_connections::table
+            .filter(graph_connections::to_account.eq(&self.address))
+            .count()
+            .get_result(&db_conn)
+            .context("Failed to count to_connections")?;
+
+        Ok(count.try_into()?)
+    }
```

Comment:
We dont have a list endpoint for wallets but if we add this needs to move to a data loader.